ci_scripts/sync_target_database.py

```diff
@@ -0,0 +1,131 @@
+"""Sync the offline target database with the online database.
+
+This utility performs the following actions:
+* Downloads the latest online target database
+* Saves the database to a local file in the mbed-targets repository
+* Creates a new branch and commits the new target database
+* Pushes the branch to the mbed-targets remote and raises a github PR as Monty Bot.
+
+NOTE:
+This script is intended to run in a CI system from mbed-targets/ci_scripts.
+
+The script assumes its location is always the directory mbed-targets/ci_scripts/. It uses this assumption to write the
+new targets.json to the correct location in the repository.
+"""
+
+import argparse
+import datetime
+import logging
+import os
+import sys
+
+from pathlib import Path
+from typing import NamedTuple
+
+from github import Github
+
+from mbed_tools_lib.exceptions import ToolsError
+from mbed_tools_lib.logging import log_exception, set_log_level
+from mbed_tools_ci_scripts.utils import git_helpers
+from mbed_targets import MbedTargets
+
+THIS_DIR = Path(__file__).absolute().parent
+
+logger = logging.getLogger()
+
+
+class PullRequestInfo(NamedTuple):
+    """Data structure containing info required to raise a Github PR."""
+
+    repo: str
+    head_branch: str
+    base_branch: str
+    subject: str
+    body: str
+
+
+def save_target_database(target_database_text: str) -> Path:
+    """Save text to data/targets.json.
+
+    Args:
+        target_database_text: text to save to mbed_targets/data/targets.json
+
+    Returns:
+        The path to data/targets.json.
+    """
+    output_file_path = Path(THIS_DIR.parent, "mbed_targets", "data", "targets.json")
```

Comment:
`THIS_DIR.parent` is somewhat nasty. I think the constant should already resolve to `TARGET_DIR`.